operator/controller/src/test/java/io/apicurio/registry/operator/it/KafkaSQLITTest.java

```diff
@@ -0,0 +1,104 @@
+package io.apicurio.registry.operator.it;
+
+import io.apicurio.registry.operator.api.v1.ApicurioRegistry3;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.rbac.ClusterRoleBinding;
+import io.fabric8.kubernetes.api.model.rbac.RoleBinding;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import io.quarkus.test.junit.QuarkusTest;
+import io.strimzi.api.kafka.model.kafka.Kafka;
+import io.strimzi.api.kafka.model.kafka.listener.ListenerStatus;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static io.apicurio.registry.operator.resource.ResourceFactory.deserialize;
+import static io.apicurio.registry.operator.resource.ResourceFactory.load;
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@QuarkusTest
+public class KafkaSQLITTest extends ITBase {
+
+    private static final Logger log = LoggerFactory.getLogger(KafkaSQLITTest.class);
+
+    @BeforeAll
+    public static void beforeAll() throws Exception {
+        ITBase.beforeAll();
+        applyStrimziResources(false);
+    }
+
+    @Test
+    void testKafkaSQLPlain() {
+        var kafka = deserialize("/k8s/examples/kafkasql/plain/ephemeral.kafka.yaml", Kafka.class);
+        client.resource(kafka).create();
+
+        await().ignoreExceptions().until(() -> {
+            var status = client.resources(Kafka.class).inNamespace(namespace).withName("my-cluster").get()
```

Comment:
Can we have a check that is not dependent on the Strimzi CR?
E.g. something that should work without code changes also for plain Kafka and RedPanda.

Reply:
We can get the status of the `ReplicaSet` of the Kafka pods, all the checks are, anyhow happening with retries and it should be more than enough to have a stable situation.

Reply:
Strimzi does not use `ReplicaSet` btw.